### PR TITLE
[cookies] optimisations

### PR DESCRIPTION
**Pull request by jrochel**

### CODE REVIEW

File: src/lib/server/eliommod_pagegen.ml [hunk 1]
```diff
@@ -114,6 +114,20 @@ let update_cookie_table ?now sitedata (ci, sci) =
 
       !data_cookies_info;
 
+    let module Expiry_tolerance = struct
+      (* Avoid cookie updates that only change the cookie
+         expiry date by a negligible amount of time. *)
+      let timeout_tolerance_factor = 0.1
```
Comment (vouillon) [inline src/lib/server/eliommod_pagegen.ml:120]: 10% is probably not negligible

**Comment by jrochel** [comment]
@vouillon @balat Ready to me merged.

File: src/lib/server/eliommod_pagegen.ml (hunk 1 above)
Comment (jrochel) [reply src/lib/server/eliommod_pagegen.ml]: @vouillon ping

File: src/lib/server/eliommod_pagegen.ml [hunk 2]
```diff
@@ -151,21 +165,24 @@ let update_cookie_table ?now sitedata (ci, sci) =
                 in
                 match oldvalue with
                   | Some (_, oldti, oldexp, oldgrp) when
-                      (oldexp = newexp &&
-                          oldti = !(newc.Eliom_common.pc_timeout) &&
-                          oldgrp = !(newc.Eliom_common.pc_session_group) &&
-                       newc.Eliom_common.pc_set_value = `None) -> Lwt.return ()
-                (* nothing to do *)
+                      Expiry_tolerance.within_tolerance_opt oldexp newexp &&
+                      Expiry_tolerance.timeout_within_tolerance oldti !(newc.Eliom_common.pc_timeout) &&
```
[the remark this answers is not in the document]
Comment (jrochel) [reply src/lib/server/eliommod_pagegen.ml]: Thanks, done!

**Pull request merged**